src/Rules/Classes/ClassConstantRule.php

```diff
@@ -63,6 +67,27 @@ public function processNode(Node $node, Scope $scope): array
 				$name = $constantString->getValue();
 				$constantNameScopes[$name] = $scope->filterByTruthyValue(new Identical($node->name, new String_($name)));
 			}
+
+			if ($this->checkNonStringableDynamicAccess) {
+				$nameTypeResult = $this->ruleLevelHelper->findTypeToCheck(
+					$scope,
+					$node->name,
+					'',
+					static fn (Type $type) => $type->isString()->yes(),
+				);
+
+				$type = $nameTypeResult->getType();
+
+				if (!$type->isString()->yes()) {
```

Comment:
```suggestion
				if (!$type instanceof ErrorType && !$type->isString()->yes()) {
```

Reply:
All calls to findTypeToCheck follow this pattern. `ErrorType` is returned if unknown classes were encountered there.